src/index.ts

```diff
@@ -164,12 +165,20 @@ class ClientApp {
         pcEnvironment?: string;
         /** The full origin (protocol, host, port) of the Genesys Cloud host environment (e.g. https://apps.mypurecloud.com).  Prefer using pcEnvironment[QueryParam] over this property. */
         pcOrigin?: string;
+        /** Name of a query param to auto parse into the gcHostOrigin; Must be valid and used with gcTargetEnvQueryParam. */
+        gcHostOriginQueryParam?: string;
+        /** Name of a query param to auto parse into the gcTargetEnv; Must be valid and used with gcHostOriginQueryParam. */
+        gcTargetEnvQueryParam?: string;
     } = {}) {
         if (cfg) {
-            if ('pcEnvironmentQueryParam' in cfg) {
+            const parsedQueryString = queryString.parse(ClientApp._getQueryString() || '');
+            if ('gcHostOriginQueryParam' in cfg || 'gcTargetEnvQueryParam' in cfg) {
+                this.assertNonEmptyString(cfg.gcHostOriginQueryParam, 'host origin query param name');
+                this.assertNonEmptyString(cfg.gcTargetEnvQueryParam, 'target env query param name');
```

Comment:
nice, thanks for asserting both of these.  We'll make it a requirement that these are always provided together.